### ISSUE TITLE
Error when chaining shortcut methods

### ISSUE THREAD

**Issue by DanielSWolf** [planttheidea/moize#179]
According to the TypeScript typings of `moize`, most methods and properties on the `Moize` type in turn return a `Moize` instance. This means that the following calls are syntactically valid:

```ts
const foo = moize.infinite.deep(fn);
const bar = moize.deep.maxSize(5)(fn);
```

**Expected behavior:**

These calls work and return memoized functions with two options applied

**Actual behavior:**

Both lines throw an error:

* TypeError: _moize.default.infinite.deep is not a function
* TypeError: _moize.default.deep.maxSize is not a function

**Comment by planttheidea** [comment]
Ah yes, this is a gap in the typing. Those methods are not intended to be chainable. I'll work on a fix for it sometimes this week.

In the meantime, I would recommend using the second parameter of options instead of chaining if you want to combine multiple options.

```ts
const foo = moize(fn, { isDeepEqual: true, maxSize: 5 });
```

**Comment by planttheidea** [comment]
I've created the PR above to resolve the issue, but going to sit on it for a day or two to review it with a fresh set of eyes.

**Comment by planttheidea** [comment]
This should be fixed as-of `6.1.3`. If you have any more issues, please let me know!
